client/my-sites/controller.js

```diff
@@ -109,6 +111,50 @@ function renderNoVisibleSites( context ) {
 	);
 }
 
+function isPathAllowedForDomainOnlySite( pathname, site ) {
+	if ( ! get( site, 'options.is_domain_only', false ) ) {
+		return true;
+	}
+
+	const urlPrefixesWhiteListForDomainOnlySite = [
+		'/domains/manage/',
+		'/checkout/',
+		'/customize/',
+		'/domains/landing-page',
```

Comment:
I think we don't want `/domains/landing-page` or `/customize/` to be in this list yet, they should also redirect.